pyang/types.py

```diff
@@ -609,7 +610,11 @@ def __call__(self, value):
         if self.schema is None:
             return None
         self._avalue.text = value
-        return self.schema.validate(self._avalue) is not self.invert_match
+        result = re.match(self.spec, value)
+        if (result is not None and result.span()[0] == 0 and
+                result.span()[1] == len(value)):
+            return not self.invert_match
```

Comment:
This:

```
result.span()[0] == 0 and result.span()[1] == len(value)
```

...could be written as:

```
result.span() == (0, len(value))
```

Reply:
Yeah you are right. But now I will revert this PR